// @retailify/ngx-mat-numberpicker/ngx-mat-numberpicker.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/material')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/common', '@angular/material'], factory) :
	(factory((global['ngx-mat-numberpicker'] = {}),global.core,global.common,global.material));
}(this, (function (exports,core,common,material) { 'use strict';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NumberPickerComponent = (function () {
    function NumberPickerComponent() {
        this.decreaseButtonDisabled = false;
        this.increaseButtonDisabled = false;
        this.hintStartText = '';
        this.hintEndText = '';
        this.disabled = false;
        this.min = 0;
        this.max = 50;
        this.onChange = new core.EventEmitter();
        this.numberPickerValue = 0;
    }
    /**
     * @return {?}
     */
    NumberPickerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.defaultValueSet();
        this.hintTextsSet();
        this.checkButtons();
    };
    /**
     * @param {?} event
     * @return {?}
     */
    NumberPickerComponent.prototype.increase = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        event.preventDefault();
        if (this.numberPickerValue < this.max) {
            this.numberPickerValue++;
            this.onChange.emit(this.numberPickerValue);
        }
        this.checkButtons();
    };
    /**
     * @param {?} event
     * @return {?}
     */
    NumberPickerComponent.prototype.decrease = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        event.preventDefault();
        if (this.getValue() > this.min) {
            this.numberPickerValue--;
            this.onChange.emit(this.numberPickerValue);
        }
        this.checkButtons();
    };
    /**
     * @return {?}
     */
    NumberPickerComponent.prototype.checkButtons = /**
     * @return {?}
     */
    function () {
        this.increaseButtonDisabled = false;
        this.decreaseButtonDisabled = false;
        if (this.getValue() <= this.min) {
            this.decreaseButtonDisabled = true;
        }
        if (this.numberPickerValue >= this.max) {
            this.increaseButtonDisabled = true;
        }
    };
    /**
     * @return {?}
     */
    NumberPickerComponent.prototype.defaultValueSet = /**
     * @return {?}
     */
    function () {
        if (this.default === null || this.default === undefined) {
            this.setValue(0);
        }
        else {
            this.setValue(this.default);
        }
    };
    /**
     * @return {?}
     */
    NumberPickerComponent.prototype.hintTextsSet = /**
     * @return {?}
     */
    function () {
        this.setHintStartText(this.starttext);
        this.setHintEndText(this.endtext);
    };
    /**
     * @return {?}
     */
    NumberPickerComponent.prototype.getValue = /**
     * @return {?}
     */
    function () {
        return this.numberPickerValue;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    NumberPickerComponent.prototype.setValue = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.numberPickerValue = value;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    NumberPickerComponent.prototype.setHintStartText = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.hintStartText = value;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    NumberPickerComponent.prototype.setHintEndText = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        this.hintEndText = value;
    };
    NumberPickerComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'ngx-mat-numberpicker',
                    template: "<span flex> <button mat-icon-button (click)=\"decrease($event)\" [disabled]=\"decreaseButtonDisabled\"><mat-icon style=\"padding-bottom: 2px; padding-left: 1px;\">remove_circle_outline</mat-icon></button> <mat-form-field flex> <input matInput #inputNumberPickerControl name=\"numberPickerInputField\" [value]=\"numberPickerValue\" style=\"text-align:center;\" [disabled]=\"disabled\" > <mat-hint align=\"start\">{{hintStartText}}</mat-hint> <mat-hint align=\"end\">{{hintEndText}}</mat-hint> </mat-form-field> <button mat-icon-button (click)=\"increase($event)\" [disabled]=\"increaseButtonDisabled\"><mat-icon style=\"padding-bottom: 2px; padding-left: 1px;\">add_circle_outline</mat-icon></button> </span> ",
                },] },
    ];
    /** @nocollapse */
    NumberPickerComponent.ctorParameters = function () { return []; };
    NumberPickerComponent.propDecorators = {
        "disabled": [{ type: core.Input },],
        "min": [{ type: core.Input },],
        "max": [{ type: core.Input },],
        "default": [{ type: core.Input },],
        "starttext": [{ type: core.Input },],
        "endtext": [{ type: core.Input },],
        "onChange": [{ type: core.Output },],
    };
    return NumberPickerComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes} checked by tsc
 */
var NumberPickerModule = (function () {
    function NumberPickerModule() {
    }
    /**
     * @return {?}
     */
    NumberPickerModule.forRoot = /**
     * @return {?}
     */
    function () {
        return {
            ngModule: NumberPickerModule,
            providers: []
        };
    };
    NumberPickerModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule,
                        material.MatButtonModule,
                        material.MatIconModule,
                        material.MatInputModule,
                    ],
                    declarations: [
                        NumberPickerComponent,
                    ],
                    exports: [
                        NumberPickerComponent,
                    ]
                },] },
    ];
    /** @nocollapse */
    NumberPickerModule.ctorParameters = function () { return []; };
    return NumberPickerModule;
}());

exports.NumberPickerModule = NumberPickerModule;
exports.NumberPickerComponent = NumberPickerComponent;

Object.defineProperty(exports, '__esModule', { value: true });

})));
